### ISSUE TITLE
Syntax highlighting - tables/fields referenced in blocks

### ISSUE THREAD

**Issue by danielbecroft** [vscode-abl/vscode-abl#184]
LanguageServerInfo: 1.16.0-SNAPSHOT-607d691 -- ClientInfo: Visual Studio Code 1.92.2 -- JVM: Oracle Corporation 17.0.6+9-LTS-190

I'm noticing some strange syntax highlighting issues when referencing table names in a block iteration (ie `do vI = 1 to Foo.Bar`). For example:

```
// scope(ttFoo.NameList): 	storage.data.table.abl meta.function.arguments.abl meta.function-call.abl meta.block.abl source.abl
// foreground: meta.block
do vIndex = 1 to num-entries(ttFoo.NameList):
end. /* */

// scope(vNameList): variable.other.abl meta.function.arguments.abl meta.function-call.abl meta.block.abl source.abl
// foreground: variable
do vIndex = 1 to num-entries(vNameList):
end. /* */
```

Everything appears to be "correct", except that the `foreground` value is getting picked up differently, resulting in different colouring. I don't know if this is related to the them, or something in the grammar, but it ends up looking like:

![image](https://github.com/user-attachments/assets/0848de4f-eab0-427d-aa09-4f775089d394)

It gets even stranger when the table/field is referenced directly in the block:

```
// scope(ttFoo): variable.other.abl meta.block.abl source.abl
// scope(MaximumValue): meta.block.abl source.abl
do vIndex = 1 to ttFoo.MaximumValue:
end. /* */

// scope(vMaximumValue): variable.other.abl meta.block.abl source.abl
do vIndex = 1 to vMaximumValue:
end. /* */
```

Here, `ttFoo` is not even being identified as a table, instead it's a variable, resulting in `MaximumValue` being parsed as the final block label. 

![image](https://github.com/user-attachments/assets/f2694cc7-1eeb-415d-9026-6adb1d69d373)

**Comment by PeterJudgeZA** [comment]
The `do i = 1 to ttfoo.field` issue is something that needs to be fixed.

In the latest version of the grammar / extension / VSCode I see the foreground value for the temp-table field as 'storage'.
```
do vIndex = 1 to num-entries(ttFoo.NameList):
end. /* */
```

**Comment by PeterJudgeZA** [comment]
This is resolved in grammar version 1.3.8
